concordium-std/src/test_infrastructure.rs

```diff
@@ -1822,6 +1824,66 @@ impl<State: StateClone<TestStateApi>> TestHost<State> {
     }
 }
 
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+use getrandom::register_custom_getrandom;
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+/// A custom function for generating random numbers.
+/// On-chain Wasm cannot use RNGs and function redirects a call to the
+/// `get_random()` primitive (host function), which is later handled by
+/// `TestHost`, where the actual random number generation happens.
+fn get_random(dest: &mut [u8]) -> Result<(), getrandom::Error> {
+    unsafe {
+        crate::prims::get_random(dest.as_mut_ptr(), dest.len() as u32);
+    }
+    Ok(())
+}
+
+#[cfg(all(feature = "wasm-test", feature = "concordium-quickcheck", target_arch = "wasm32"))]
+// Register our own custom random number generation function, so all the calls,
+// that depend on `getrandom` (like `from_entropy()`) will call our function
+// instead. This is only relevant for Wasm, where RNGs are not
+// supported.
+register_custom_getrandom!(get_random);
+
+#[cfg(all(feature = "concordium-quickcheck", target_arch = "wasm32"))]
+/// A customized QuickCheck test runner used for on-chain wasm code.
+/// Adds support for reporting errors using the primitives available when
+/// running Wasm code.
+pub fn concordium_qc<A>(f: A)
+where
+    A: Testable, {
```

Comment:
```suggestion
pub fn concordium_qc<A: Testable>(f: A)
```

That's the style we generally prefer for short constraints.